Fixed DATEADD() numeric field overflow error

ZhenyeLee opened the pull request:
Fixed DATEADD() numeric field overflow error

Task: BABEL-3436
Signed-off-by: Zhenye Li <zhenye@amazon.com>

### Description

Fixed numeric field overflow error when casting int to numeric number. It was CAST(num AS DECIMAL(2,0)), and it failed when int number is larger than 99. Now it is CAST(num AS DECIMAL(20,10)), which can handle 10bits Integer.Add MVU file and disable the microsecond unit which is not support in SQL Server
 

### Issues Resolved

Fixed numeric field overflow error when casting int to numeric number and disable the microsecond unit which is not support in SQL Server


### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
Fixed RDSManfred-Babel Extension tests

Fixed numeric field overflow error when casting int to numeric number. It was CAST(num AS DECIMAL(2,0)), and it failed when int number is larger than 99. Now it is CAST(num AS DECIMAL(20,10)), which can handle 10bits Integer.Add MVU file and disable the microsecond unit which is not support in SQL Server

Task: MANFRED-13687
Signed-off-by: Zhenye Li <zhenye@amazon.com>

Commit:
Fixed test file

Task: MANFRED-13687
Signed-off-by: Zhenye Li <zhenye@amazon.com>

ZhenyeLee committed:
Fix conflict

ZhenyeLee committed:
Update babelfishpg_tsql--2.1.0--2.2.0.sql

surendravishnoi commented on test/JDBC/input/BABEL-3360-vu-prepare.sql:
```diff
@@ -0,0 +1,19 @@
+select * from BABEL_3360_vu_prepare_v3;
+go
+
+create view BABEL_3360_vu_prepare_v4 as (select dateadd(microsecond, 56, null));
```
we should also call the functions from function/procedure

ZhenyeLee committed:
Update BABEL-3360-vu-prepare.out

ZhenyeLee committed:
Update BABEL-3360-vu-verify.out

ZhenyeLee committed:
Update BABEL-3360-vu-prepare.sql

ZhenyeLee committed:
Update BABEL-3360-vu-verify.sql

ZhenyeLee committed:
Update BABEL-3360-vu-verify.sql

ZhenyeLee committed:
Update BABEL-3360-vu-verify.out

ZhenyeLee committed:
Merge branch 'babelfish-for-postgresql:BABEL_2_X_DEV' into JIRA-babel-3360

ZhenyeLee committed:
Update babel_datetime-vu-verify.out

ZhenyeLee committed:
Update BABEL-3360-vu-prepare.out

ZhenyeLee committed:
Update latest__verification_cleanup__13_4__babel_datetime-vu-verify.out

ZhenyeLee committed:
Update latest__verification_cleanup__13_5__babel_datetime-vu-verify.out

ZhenyeLee committed:
Update latest__verification_cleanup__13_6__babel_datetime-vu-verify.out

ZhenyeLee committed:
Update latest__verification_cleanup__14_3__babel_datetime-vu-verify.out

ZhenyeLee committed:
Merge branch 'babelfish-for-postgresql:BABEL_2_X_DEV' into JIRA-babel-3360

ZhenyeLee committed:
Update expected_create.out

ZhenyeLee committed:
Update expected_dependency.out

ZhenyeLee committed:
Update expected_dependency.out

ZhenyeLee committed:
Update expected_create.out

surendravishnoi reviewed (commented):
PR subject should have more context about issue fixed. Current one is too generic. Do not put any CR links here. Also, provide complete Jira id with manfred/babel prefix.

ZhenyeLee commented:
> PR subject should have more context about issue fixed. Current one is too generic. Do not put any CR links here. Also, provide complete Jira id with manfred/babel prefix.

I updated the PR subject and deleted CR links. But since MANFRED is out internal project, it that okay to include in Github?

surendravishnoi commented:
> > PR subject should have more context about issue fixed. Current one is too generic. Do not put any CR links here. Also, provide complete Jira id with manfred/babel prefix.
> 
> I updated the PR subject and deleted CR links. But since MANFRED is out internal project, it that okay to include in Github?

we should not include manfred here. You can open a babel jira, link it with manfred Jira and put babel Jira number here (babel-num)

ZhenyeLee commented:
> > > PR subject should have more context about issue fixed. Current one is too generic. Do not put any CR links here. Also, provide complete Jira id with manfred/babel prefix.
> > 
> > 
> > I updated the PR subject and deleted CR links. But since MANFRED is out internal project, it that okay to include in Github?
> 
> we should not include manfred here. You can open a babel jira, link it with manfred Jira and put babel Jira number here (babel-num)

Updated PR Description with BABEL JIRA number, please take a look. Thanks!

surendravishnoi commented on test/JDBC/upgrade/latest/schedule:
```diff
@@ -89,6 +89,7 @@ BABEL-3249
 BABEL-383
 BABEL-405
 BABEL-PROCID
+BABEL-3360
```
why only to latest schedule file? we should add it to all other upgrade sources like 13.4, 13.5 up to 14.3

surendravishnoi reviewed (commented):
Pleas resolve merge conflict

ZhenyeLee committed:
Resolve conflict

ZhenyeLee commented:
Created a new PR https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/497
